### ISSUE TITLE
Support balance assertion tolerance syntax (`~`)

### ISSUE THREAD

**Issue by blais** [jcornaz/beancount-parser#70]
Thanks for addressing all the other issues Jonathan.
Updating and continuing on, I see the precision syntax fail:
```
2013-09-10 balance Assets:US:Vanguard                                   305.205 ~ 0.002 RGAGX
```

**Comment by jcornaz** [comment]
Thanks for the report, I didn't know that syntax. 

I usually refer to this document: https://beancount.github.io/docs/beancount_language_syntax.html and it looks like doesn't mention it. 

But of course, I'd like this parser to accept any valid beancount syntax, so the `~` precision syntax should be supported.

**Comment by blais** [comment]
Great! 
Id like to build something and learn some Rust in the process using your parser.

**Comment by blais** [comment]
Local tolerance parsing looks like this in bison:```
amount_tolerance:
  number_expr currency
    {
        BUILDY(DECREF($1, $2),
               $$.pyobj1, "amount", "OO", $1, $2);
        $$.pyobj2 = Py_None;
        Py_INCREF(Py_None);
    }
  | number_expr TILDE number_expr currency
    {
        BUILDY(DECREF($1, $4),
               $$.pyobj1, "amount", "OO", $1, $4);
        $$.pyobj2 = $3;
    }
```

**Comment by blais** [comment]
I added some text int he docs: 
https://docs.google.com/document/d/1wAMVrKIA2qtRGmoVDSUBJGmYZSygUaR0uOMW1GV3YE0/edit#heading=h.5jx4fqseqk64

**Comment by jcornaz** [comment]
This is fixed in version [2.1.0](https://github.com/jcornaz/beancount-parser/releases/tag/v2.1.0)

**Comment by blais** [comment]
Thank you!  :-)
I now confirm this parser is able to grok my entire gnarly personal Beancount file.
